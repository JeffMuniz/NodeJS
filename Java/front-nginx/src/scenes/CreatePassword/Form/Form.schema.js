import { errorMessages } from "@enums";
import FormSchemaValidator from "src/modules/Form/FormSchemaValidator";

export const strongPassword = RegExp(
  /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,15}$/,
);

const fields = {
  password: FormSchemaValidator.string()
    .matches(strongPassword, errorMessages.StrongPassword)
    .required(errorMessages.FieldRequired)
    .min(5)
    .max(15),
  confirmPassword: FormSchemaValidator.string()
    .oneOf([FormSchemaValidator.ref("password")], errorMessages.MatchPassword)
    .required(errorMessages.FieldRequired)
    .matches(strongPassword, errorMessages.StrongPassword),
};

export default FormSchemaValidator.object().shape(fields);
