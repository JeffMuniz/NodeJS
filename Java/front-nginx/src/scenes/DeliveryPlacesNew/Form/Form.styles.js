import styled from "styled-components";
import { Row } from "react-styled-flexboxgrid";

import { red } from "@colors";

export const RowContent = styled(Row)`
  flex-direction: column;
  min-height: 250px;
`;

export const RowFooter = styled(Row)`
  margin-bottom: 116px;
`;

export const LabelUpload = styled.label`
  font-weight: 700;
  line-height: normal;
  font-size: 14px;
  letter-spacing: -0.3px;
  text-transform: uppercase;
  color: ${red};
  display: inline-flex;
  margin: 0;
  border: none;
  cursor: pointer;
  background: transparent;
`;

export const InputFile = styled.input`
  display: none;
`;
