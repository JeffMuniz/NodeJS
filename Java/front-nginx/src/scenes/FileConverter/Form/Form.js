import React, { Fragment } from "react";
import { Row, Col } from "react-styled-flexboxgrid";
import {
  func,
  shape,
  string,
  bool,
  number,
  oneOfType,
  arrayOf,
} from "prop-types";
import { If } from "@utils";

import {
  Alert,
  Button,
  SvgIcon,
  FormGroup,
  TextInput,
  SingleDatePicker,
} from "@common";
import { dateHourFormats, buttonTypes } from "@enums";
import DateManager from "src/modules/DateManager/DateManager";

import {
  RowContent,
  RowFooter,
  LabelUpload,
  Title,
  fileInput,
  DatePlaceholder,
  CancelButtonContainer,
} from "./Form.styles";

const Form = ({
  handleSubmit,
  handleChangeFile,
  handleChange,
  handleBackButton,
  isFileSelected,
  isFileValid,
  file,
  values,
  errors,
  requestError,
  openFileDialog,
  clientCodes,
  fileInputRef,
  showUpload,
  showData,
  submitValidation,
  showDateField,
  setFieldValue,
  onChangeFocus,
  datePickerFocused,
}) => (
  <Fragment>
    <If test={showUpload}>
      <Fragment>
        <Row>
          <Col xs={12}>
            <LabelUpload id="file_button" onClick={openFileDialog}>
              <SvgIcon name="import" />
              Anexar TXT
            </LabelUpload>
            <input
              ref={fileInputRef}
              id="file"
              name="file"
              type="file"
              style={fileInput}
              value={values.file}
              onChange={event => handleChangeFile(event)}
            />
          </Col>
        </Row>
        <RowContent>
          <Col xs={12}>
            {isFileSelected() && isFileValid && (
              <Alert
                id="on_feedback_file"
                title={file.name}
                image="success"
                type="success"
              />
            )}

            {isFileSelected() && !isFileValid && (
              <Alert
                id="on_feedback_file"
                title="Ocorreu um erro."
                image="cancelledStatus"
                description={
                  requestError || (
                    <div>
                      Confira os requisitos: formato <b>.txt</b>, e{" "}
                      <b>todos os campos precisam estar preenchidos</b> e
                      adicione novamente.
                    </div>
                  )
                }
                type="error"
              />
            )}
          </Col>
        </RowContent>
        <RowFooter center="xs">
          <Col xs={4}>
            <Button
              id="fc_btn_submit_validate"
              type="button"
              disabled={!isFileValid}
              value="Importar arquivo"
              onPress={submitValidation}
            />
          </Col>
        </RowFooter>
      </Fragment>
    </If>
    <If test={showDateField}>
      <Fragment>
        <form onSubmit={handleSubmit}>
          <Row>
            <Title>Dados para Conversão</Title>
          </Row>
          <Row center="xs">
            <FormGroup fieldId="dataPedido" label="Informe a data do crédito">
              <div>
                <DatePlaceholder id="dataPedidoButton" onClick={onChangeFocus}>
                  {values.dataPedido
                    ? DateManager(values.dataPedido).format(
                        dateHourFormats.longDateSlash,
                      )
                    : dateHourFormats.longDateSlash}
                </DatePlaceholder>
                <If test={datePickerFocused}>
                  <SingleDatePicker
                    id="dataPedido"
                    onDateChange={({ date }) =>
                      setFieldValue(
                        "dataPedido",
                        DateManager(date).format(dateHourFormats.longDateUS),
                      )
                    }
                    numberOfMonths={2}
                    renderCalendarInfo={() => (
                      <CancelButtonContainer>
                        <Button
                          buttonType={buttonTypes.light}
                          value="Cancelar"
                        />
                      </CancelButtonContainer>
                    )}
                    onCloseDatePicker={onChangeFocus}
                    autoFocus
                  />
                </If>
              </div>
            </FormGroup>
          </Row>
          <RowFooter center="xs">
            <Col xs={3}>
              <Button
                id="fc_btn_back"
                buttonType="light"
                value="Voltar"
                onPress={handleBackButton}
              />
            </Col>
            <Col xs={3}>
              <Button
                id="fc_btn_submit_convert"
                type="submit"
                disabled={!isFileValid}
                value="Converter Arquivo"
              />
            </Col>
          </RowFooter>
        </form>
      </Fragment>
    </If>
    <If test={showData}>
      <Fragment>
        <form onSubmit={handleSubmit}>
          <Row>
            <Title>Dados para Conversão</Title>
          </Row>
          {clientCodes.map((code, index) => (
            <Row center="xs" key={code}>
              <Col xs={4}>
                <FormGroup fieldId="fc_codigoCliente" label="Código Cliente">
                  <TextInput
                    id={`codigoCliente_${index}`}
                    name={`codigoCliente_${index}`}
                    value={code}
                    maskChar=""
                    placeholder="Código Cliente"
                  />
                </FormGroup>
              </Col>
              <Col xs={4}>
                <FormGroup
                  fieldId="fc_cnpj"
                  label="CNPJ"
                  error={errors && errors[`cnpjCliente_${index}`]}
                >
                  <TextInput
                    id={`cnpjCliente_${index}`}
                    name={`cnpjCliente_${index}`}
                    onChange={handleChange}
                    value={values[`cnpjCliente_${index}`]}
                    maskType="cnpj"
                    maskChar=""
                    placeholder="CNPJ"
                  />
                </FormGroup>
              </Col>
            </Row>
          ))}

          <RowFooter center="xs">
            <Col xs={3}>
              <Button
                id="fc_btn_back"
                buttonType="light"
                value="Voltar"
                onPress={handleBackButton}
              />
            </Col>
            <Col xs={3}>
              <Button
                id="fc_btn_submit_convert"
                type="submit"
                disabled={!isFileValid}
                value="Converter Arquivo"
              />
            </Col>
          </RowFooter>
        </form>
      </Fragment>
    </If>
  </Fragment>
);

Form.defaultProps = {
  file: {},
  values: {},
  requestError: "",
  isFileValid: false,
  isFileSelected: () => false,
  setFieldValue: () => null,
  datePickerFocused: false,
  onChangeFocus: () => null,
};

Form.propTypes = {
  setFieldValue: func,
  datePickerFocused: bool,
  onChangeFocus: func,
  handleBackButton: func.isRequired,
  openFileDialog: func.isRequired,
  fileInputRef: shape({}).isRequired,
  handleSubmit: func.isRequired,
  handleChangeFile: func.isRequired,
  handleChange: func.isRequired,
  submitValidation: func.isRequired,
  showUpload: bool.isRequired,
  showData: bool.isRequired,
  showDateField: bool.isRequired,
  isFileSelected: func,
  isFileValid: bool,
  clientCodes: arrayOf(
    shape({
      codigo: string,
      cnpjCliente: string,
    }),
  ).isRequired,
  errors: shape({
    cnpj: string,
  }).isRequired,
  file: shape({
    name: string,
  }),
  values: shape({
    file: string,
  }),
  requestError: oneOfType([
    string,
    shape({
      error: shape({
        code: number,
        message: string,
        type: string,
      }),
    }),
  ]),
};

export default Form;
