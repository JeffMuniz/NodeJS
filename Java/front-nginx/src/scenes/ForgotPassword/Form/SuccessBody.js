import React, { Fragment } from "react";
import { Row } from "react-flexbox-grid";
import { func } from "prop-types";
import { Button } from "@common";

import {
  StyledColSuccess,
  LittleLine,
  LastCol,
  RowButton,
} from "./SuccessBody.styles";

const SuccessBody = ({ returnToLogin }) => (
  <Fragment>
    <Row center="md">
      <StyledColSuccess>
        Enviamos um e-mail para a criação da sua nova senha.
      </StyledColSuccess>
    </Row>
    <LittleLine />
    <Row center="md">
      <LastCol>
        Caso exista algum problema com seu email, entre em contato com o usuário
        administrador da sua empresa.
      </LastCol>
    </Row>
    <RowButton center="md">
      <Button
        id="pwd_btn_goback"
        buttonType="actionButton"
        onPress={returnToLogin}
        title="Voltar"
        value="voltar"
      />
    </RowButton>
  </Fragment>
);

SuccessBody.propTypes = {
  returnToLogin: func.isRequired,
};

export default SuccessBody;
