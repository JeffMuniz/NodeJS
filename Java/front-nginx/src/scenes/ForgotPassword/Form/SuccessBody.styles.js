import styled from "styled-components";
import { Row, Col } from "react-flexbox-grid";
import { grey, darkGrey } from "@colors";

export const LittleLine = styled.div`
  background: ${darkGrey};
  height: 2px;
  border-radius: 1px;
  margin: 26px auto 52px;
  width: 80px;
`;

export const StyledColSuccess = styled(Col)`
  display: flex;
  justify-content: center;
  font-style: normal;
  font-weight: 300;
  font-size: 20px;
  line-height: 30px;
  color: ${grey};
`;

export const LastCol = styled(StyledColSuccess)`
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 26px;
  letter-spacing: 0.45px;
`;

export const RowButton = styled(Row)`
  margin-top: 133px;
  margin-bottom: 75px;
`;

export const Br = styled.br``;
