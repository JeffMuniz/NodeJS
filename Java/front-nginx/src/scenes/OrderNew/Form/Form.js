import React, { Fragment } from "react";
import { Row, Col } from "react-styled-flexboxgrid";
import {
  func,
  shape,
  string,
  bool,
  number,
  oneOfType,
  arrayOf,
} from "prop-types";

import {
  Alert,
  Button,
  SvgIcon,
  FormGroup,
  TextInput,
  SingleDatePicker,
} from "@common";
import { If } from "@utils";
import { dateHourFormats, buttonTypes } from "@enums";
import DateManager from "src/modules/DateManager/DateManager";

import {
  RowContent,
  RowFooter,
  LabelUpload,
  fileInput,
  Title,
  InputContainer,
  DatePlaceholder,
  CancelButtonContainer,
} from "./Form.styles";

const Form = ({
  handleSubmit,
  handleChangeFile,
  isFileSelected,
  goToPageOrders,
  isFileValid,
  isFileSizeValid,
  file,
  values,
  errors,
  requestError,
  openFileDialog,
  fileInputRef,
  showDateField,
  showConverterForm,
  handleChange,
  clientCodes = [],
  conversionError,
  clearError,
  setFieldValue,
  onChangeFocus,
  datePickerFocused,
}) => (
  <form onSubmit={handleSubmit}>
    <Fragment>
      <Row>
        <Col xs={12}>
          <LabelUpload id="file_button" onClick={openFileDialog}>
            <SvgIcon name="import" />
            Anexar arquivo
          </LabelUpload>
          <input
            ref={fileInputRef}
            id="file"
            name="file"
            type="file"
            style={fileInput}
            value={values.file}
            onChange={event => handleChangeFile(event)}
          />
        </Col>
      </Row>
      <RowContent>
        <Col xs={12}>
          <If test={conversionError !== null}>
            <Alert
              id="conversionError"
              title="Ocorreu um erro."
              image="cancelledStatus"
              description={conversionError}
              type="error"
            />
          </If>
          <If test={isFileSelected() && isFileValid && !conversionError}>
            <Alert
              id="on_feedback_file"
              title={file.name}
              image="success"
              type="success"
            />
          </If>
          <If test={isFileSelected() && !isFileValid}>
            <Alert
              id="on_feedback_file"
              title="Ocorreu um erro."
              image="cancelledStatus"
              description={
                requestError ||
                (isFileSizeValid ? (
                  <div>
                    Confira os requisitos: formatos <b>.xls, .xlsx ou .txt</b>,
                    e <b>todos os campos precisam estar preenchidos</b> e
                    adicione novamente.
                  </div>
                ) : (
                  <div>O arquivo importado possui tamanho superior a 10MB.</div>
                ))
              }
              type="error"
            />
          </If>
        </Col>
        <If test={showDateField}>
          <Fragment>
            <Row>
              <Title>Dados para Conversão</Title>
            </Row>
            <Row>
              <InputContainer>
                <FormGroup
                  fieldId="dataPedido"
                  label="Informe a data do crédito"
                >
                  <div>
                    <DatePlaceholder
                      id="dataPedidoButton"
                      onClick={onChangeFocus}
                    >
                      {values.dataPedido
                        ? DateManager(values.dataPedido).format(
                            dateHourFormats.longDateSlash,
                          )
                        : dateHourFormats.longDateSlash}
                    </DatePlaceholder>
                    <If test={datePickerFocused}>
                      <SingleDatePicker
                        id="dataPedido"
                        onDateChange={({ date }) =>
                          setFieldValue(
                            "dataPedido",
                            DateManager(date).format(
                              dateHourFormats.longDateUS,
                            ),
                          )
                        }
                        numberOfMonths={2}
                        renderCalendarInfo={() => (
                          <CancelButtonContainer>
                            <Button
                              buttonType={buttonTypes.light}
                              value="Cancelar"
                            />
                          </CancelButtonContainer>
                        )}
                        onCloseDatePicker={onChangeFocus}
                        autoFocus
                      />
                    </If>
                  </div>
                </FormGroup>
              </InputContainer>
            </Row>
          </Fragment>
        </If>
        <If test={showConverterForm}>
          <Fragment>
            <Title>Dados para Conversão</Title>
            {(clientCodes || []).map((code, index) => (
              <Row center="xs" key={code}>
                <Col xs={4}>
                  <FormGroup fieldId="fc_codigoCliente" label="Código Cliente">
                    <TextInput
                      id={`codigoCliente_${index}`}
                      name={`codigoCliente_${index}`}
                      value={code}
                      maskChar=""
                      placeholder="Código Cliente"
                    />
                  </FormGroup>
                </Col>
                <Col xs={4}>
                  <FormGroup
                    fieldId="fc_cnpj"
                    label="CNPJ"
                    error={errors && errors[`cnpjCliente_${index}`]}
                  >
                    <TextInput
                      id={`cnpjCliente_${index}`}
                      name={`cnpjCliente_${index}`}
                      onChange={handleChange}
                      value={values[`cnpjCliente_${index}`]}
                      maskType="cnpj"
                      maskChar=""
                      placeholder="CNPJ"
                    />
                  </FormGroup>
                </Col>
              </Row>
            ))}
          </Fragment>
        </If>
      </RowContent>
      <RowFooter center="xs">
        <Col xs={3}>
          <If
            test={
              (isFileSelected() && !isFileValid) || conversionError !== null
            }
          >
            <Button
              id="lg_btn_goback"
              type="button"
              buttonType="light"
              value="Voltar"
              onClick={clearError}
              action="Clicou em 'Voltar' na página 'Novo Pedido'"
            />
          </If>
          <If
            test={
              (!isFileSelected() && !isFileValid) ||
              (isFileValid && conversionError === null)
            }
          >
            <Button
              id="lg_btn_goback"
              type="button"
              buttonType="light"
              value="Voltar"
              onClick={goToPageOrders}
              action="Clicou em 'Voltar' na página 'Novo Pedido'"
            />
          </If>
        </Col>
        <Col xs={3}>
          <Button
            id="lg_btn_submit"
            type="submit"
            disabled={!isFileValid || conversionError !== null}
            value="Enviar pedido"
            action="Clicou em 'Enviar Pedido' na página 'Novo Pedido'"
          />
        </Col>
      </RowFooter>
    </Fragment>
  </form>
);

Form.defaultProps = {
  file: {},
  values: {},
  requestError: "",
  conversionError: null,
  showDateField: false,
  showConverterForm: false,
  clientCodes: [],
  isFileValid: false,
  isFileSizeValid: false,
  isFileSelected: () => false,
  goToPageOrders: () => null,
  setFieldValue: () => null,
  datePickerFocused: false,
  onChangeFocus: () => null,
};

Form.propTypes = {
  showDateField: bool,
  showConverterForm: bool,
  setFieldValue: func,
  datePickerFocused: bool,
  onChangeFocus: func,
  clearError: func.isRequired,
  clientCodes: arrayOf(
    shape({
      codigo: string,
      cnpjCliente: string,
    }),
  ),
  errors: shape({
    cnpj: string,
  }).isRequired,
  handleChange: func.isRequired,
  conversionError: string,
  openFileDialog: func.isRequired,
  fileInputRef: shape({}).isRequired,
  handleSubmit: func.isRequired,
  handleChangeFile: func.isRequired,
  isFileSelected: func,
  goToPageOrders: func,
  isFileValid: bool,
  isFileSizeValid: bool,
  file: shape({
    name: string,
  }),
  values: shape({
    file: string,
  }),
  requestError: oneOfType([
    string,
    shape({
      error: shape({
        code: number,
        message: string,
        type: string,
      }),
    }),
  ]),
};

export default Form;
