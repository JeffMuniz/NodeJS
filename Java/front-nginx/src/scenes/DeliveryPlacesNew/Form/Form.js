import React, { Fragment } from "react";
import { Col } from "react-styled-flexboxgrid";
import { func, shape, string, bool, number, oneOfType } from "prop-types";

import { Alert, Button, SvgIcon } from "@common";
import { grey } from "@colors";
import { If } from "@utils";

import { RowContent, RowFooter, LabelUpload, InputFile } from "./Form.styles";

const Form = ({
  handleSubmit,
  handleChangeFile,
  isFileSelected,
  goToDeliveryPlaces,
  isFileValid,
  file,
  values,
  requestError,
}) => {
  const fileUploadButton = () => (
    <Fragment>
      <LabelUpload
        id="file_button"
        htmlFor="file"
        icon="upload"
        title="Reenviar"
      >
        <SvgIcon name="upload" fill={grey} />
      </LabelUpload>
      <InputFile
        id="file"
        name="file"
        type="file"
        value={values.file}
        onChange={event => handleChangeFile(event)}
      />
    </Fragment>
  );

  return (
    <form onSubmit={handleSubmit}>
      <Fragment>
        <RowContent>
          <Col xs={12}>
            <If test={isFileSelected() && isFileValid}>
              <Alert
                id="on_feedback_file"
                title={file.name}
                image="check"
                size={35}
                action={fileUploadButton()}
                type="success"
                description={
                  <p>
                    Planilha validada. Pode continuar para processar os dados.
                  </p>
                }
              />
            </If>

            <If test={isFileSelected() && !isFileValid}>
              <Alert
                id="on_feedback_file"
                title="Ocorreu um erro."
                image="cancelledStatus"
                size={35}
                action={fileUploadButton()}
                description={
                  requestError || (
                    <p>
                      Confira os requisitos: formatos <b>.xls ou .xlsx</b>, e
                      &nbsp;
                      <b>todos os campos precisam estar preenchidos</b> e
                      adicione novamente.
                    </p>
                  )
                }
                type="error"
              />
            </If>
          </Col>
        </RowContent>

        <RowFooter center="xs">
          <Col xs={3}>
            <Button
              id="lg_btn_goback"
              type="button"
              buttonType="light"
              value="Voltar"
              onClick={goToDeliveryPlaces}
            />
          </Col>
          <Col xs={3}>
            <Button
              id="lg_btn_submit"
              type="submit"
              disabled={!isFileValid}
              value="Enviar local de entrega"
            />
          </Col>
        </RowFooter>
      </Fragment>
    </form>
  );
};

Form.propTypes = {
  handleSubmit: func.isRequired,
  handleChangeFile: func.isRequired,
  isFileSelected: func,
  goToDeliveryPlaces: func,
  isFileValid: bool,
  file: oneOfType([
    string,
    shape({
      name: string,
    }),
  ]),
  values: shape({
    file: string,
  }),
  requestError: oneOfType([
    string,
    shape({
      error: shape({
        code: number,
        message: string,
        type: string,
      }),
    }),
  ]),
};

Form.defaultProps = {
  file: {},
  values: {},
  requestError: "",
  isFileValid: false,
  isFileSelected: () => false,
  goToDeliveryPlaces: () => null,
};

export default Form;
