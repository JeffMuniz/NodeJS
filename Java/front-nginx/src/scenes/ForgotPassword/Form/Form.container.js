import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { string, func, shape, oneOfType, instanceOf } from "prop-types";

import { requestStatus } from "@enums";
import { If } from "@utils";

import ResponseError from "src/common/entities/ResponseError";
import FormWrapper from "src/modules/Form/Form";
import * as userActions from "src/redux/modules/user/actions/userActions";
import * as SessionActions from "src/redux/modules/session/actions/session";

import FormSchema from "./Form.schema";
import Form from "./Form";
import Success from "./SuccessBody";

const initialValues = {
  forgotCpf: "",
  email: "",
};

export class ForgetPasswordContainer extends Component {
  state = {
    forgotPasswordError: null,
    emailForgotStatus: false,
    cpfForgotStatus: false,
    robot: true,
  };

  static getDerivedStateFromProps(nextProps) {
    if (nextProps.forgotPasswordData.requestStatus === requestStatus.success) {
      return { forgotPasswordSuccess: true };
    }

    if (nextProps.forgotPasswordData.requestStatus === requestStatus.error) {
      return {
        forgotPasswordError: nextProps.forgotPasswordData.error.message,
        forgotPasswordSuccess: false,
        emailForgotStatus: true,
        cpfForgotStatus: true,
      };
    }

    return {
      forgotPasswordError: null,
      forgotPasswordSuccess: false,
      emailForgotStatus: false,
      cpfForgotStatus: false,
    };
  }

  componentDidMount() {
    const { resetAuth } = this.props;
    resetAuth();
  }

  componentWillUnmount() {
    const { resetForgotPassword } = this.props;

    this.setState({
      forgotPasswordError: null,
      emailForgotStatus: false,
      cpfForgotStatus: false,
      robot: true,
    });

    resetForgotPassword();
  }

  onVerifyRecaptcha = () => {
    this.setState({ robot: false });
  };

  onExpiredRecaptcha = () => {
    this.setState({ robot: true });
  };

  changeInputStatus = input => () => {
    this.setState({ [input]: false });
  };

  returnToLogin = () => {
    const {
      history: { goBack },
    } = this.props;

    goBack();
  };

  submit = async ({ forgotCpf: cpf, email }) => {
    const { getCode, forgotPasswordAction, toggleLoading } = this.props;
    try {
      toggleLoading();
      await Promise.all([
        await getCode(),
        await forgotPasswordAction({ cpf, email }),
      ]);
      toggleLoading();
    } catch (error) {
      toggleLoading();
    }
  };

  render() {
    const { forgotPasswordData } = this.props;
    const {
      forgotPasswordError,
      emailForgotStatus,
      cpfForgotStatus,
      forgotPasswordSuccess,
      robot,
    } = this.state;

    return (
      <Fragment>
        <If test={!forgotPasswordSuccess}>
          <div id="ForgotPwdForm">
            <FormWrapper
              initialValues={initialValues}
              onSubmit={this.submit}
              validationSchema={FormSchema}
              enableReinitialize
              render={props => (
                <Form
                  requestError={forgotPasswordError}
                  isLoading={
                    forgotPasswordData.requestStatus === requestStatus.loading
                  }
                  emailForgotStatus={emailForgotStatus}
                  cpfForgotStatus={cpfForgotStatus}
                  changeInputStatus={this.changeInputStatus}
                  returnToLogin={this.returnToLogin}
                  onVerifyRecaptcha={this.onVerifyRecaptcha}
                  onErrored={this.onExpiredRecaptcha}
                  isRobot={robot}
                  {...props}
                />
              )}
            />
          </div>
        </If>
        <If test={forgotPasswordSuccess}>
          <div id="ForgotPwdSuccess">
            <Success returnToLogin={this.returnToLogin} />
          </div>
        </If>
      </Fragment>
    );
  }
}

export function mapStateToProps({ user: { forgotPassword } }) {
  return {
    forgotPasswordData: forgotPassword,
  };
}

const mapDispatchToProps = {
  forgotPasswordAction: userActions.forgotPassword,
  resetForgotPassword: userActions.resetForgotPassword,
  resetAuth: SessionActions.resetAuth,
  getCode: SessionActions.getCode,
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(ForgetPasswordContainer),
);

ForgetPasswordContainer.propTypes = {
  forgotPasswordAction: func.isRequired,
  resetAuth: func.isRequired,
  getCode: func.isRequired,
  forgotPasswordData: shape({
    requestStatus: string.isRequired,
    error: oneOfType([
      instanceOf(ResponseError),
      shape({
        message: string,
      }),
    ]),
  }),
  resetForgotPassword: func.isRequired,
  toggleLoading: func.isRequired,
};

ForgetPasswordContainer.defaultProps = {
  forgotPasswordData: {},
};
