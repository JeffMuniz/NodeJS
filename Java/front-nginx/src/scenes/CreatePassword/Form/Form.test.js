import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Form from "./Form";

jest.mock("@common", () => ({
  Button: "Button",
  ErrorText: "ErrorText",
  TextInput: "TextInput",
  CheckBox: "CheckBox",
  RulePassword: "RulePassword",
  FormGroup: "FormGroup",
}));

let props = {};

beforeEach(() => {
  props = {
    handleSubmit: jest.fn(),
    handleChange: jest.fn(),
    handleBlur: jest.fn(),
    values: {},
    touched: {},
    errors: {},
    isValid: false,
    downloadTOSFile: jest.fn(),
  };
});

describe("FirstAccess - Form", () => {
  it("Should render form", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<Form {...props} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });

  it("Should render it when error is given", () => {
    // given
    const renderer = new ShallowRenderer();
    const errors = {
      password: "Inválida",
      confirmPassword: "Inválida",
    };
    const touched = {
      password: true,
      confirmPassword: true,
    };

    // when
    renderer.render(
      <Form
        {...props}
        touched={touched}
        errors={errors}
        requestError="Error"
      />,
    );
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
