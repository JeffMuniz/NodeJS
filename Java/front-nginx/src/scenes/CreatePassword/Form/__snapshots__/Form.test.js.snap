// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstAccess - Form Should render form 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Row
    center="xs"
  >
    <Styled(Col)>
      <strong>
        lembre-se que sua senha precisa ter ao menos:
      </strong>
    </Styled(Col)>
    <styled.ul>
      <RulePassword
        error={undefined}
        id="min-caracter"
        rule={/\\.\\{8,15\\}/}
        text="Mínimo 8 e máximo 15 caracteres"
        value={undefined}
      />
      <RulePassword
        error={undefined}
        id="um-maiscula"
        rule={/\\^\\(\\?=\\.\\*\\?\\[A-Z\\]\\)/}
        text="uma letra maiúscula"
        value={undefined}
      />
    </styled.ul>
    <styled.ul>
      <RulePassword
        error={undefined}
        id="um-minuscula"
        rule={/\\(\\?=\\.\\*\\?\\[a-z\\]\\)/}
        text="Uma letra minúscula"
        value={undefined}
      />
      <RulePassword
        error={undefined}
        id="carcter-especial"
        rule={/\\(\\?=\\.\\*\\?\\[#\\?!@\\$%\\^&\\*-\\]\\)/}
        text="Um caractere especial (Exemplo: @ # * & % $)"
        value={undefined}
      />
      <RulePassword
        error={undefined}
        id="um-number"
        rule={/\\(\\?=\\.\\*\\?\\[0-9\\]\\)/}
        text="Um número"
        value={undefined}
      />
    </styled.ul>
  </Row>
  <styled.FormGroup
    error={undefined}
    fieldId="cp_password"
    label="Criar uma Senha"
    showSuccess={undefined}
  >
    <TextInput
      eye={true}
      id="password"
      maskType={undefined}
      maxLength={15}
      name="password"
      onBlur={[MockFunction]}
      onChange={[Function]}
      passwordView={[Function]}
      placeholder="Insira aqui sua senha"
      value={undefined}
    />
  </styled.FormGroup>
  <styled.FormGroup
    error={undefined}
    fieldId="cp_confirm_password"
    label="Confirmar senha"
    showSuccess={undefined}
  >
    <TextInput
      error={undefined}
      eye={true}
      id="confirmPassword"
      maskType={undefined}
      maxLength={15}
      name="confirmPassword"
      onBlur={[MockFunction]}
      onChange={[Function]}
      passwordView={[Function]}
      placeholder="Repita a sua senha"
      value={undefined}
    />
  </styled.FormGroup>
  <Styled(Row)>
    <Col
      xs={12}
    />
  </Styled(Row)>
  <Styled(Row)
    center="md"
  >
    <Col
      md={12}
      xs={12}
    >
      <Styled(Row)>
        <UNDEFINED
          hl="pt-BR"
          onChange={undefined}
          onErrored={undefined}
          onExpired={undefined}
          sitekey=""
        />
        <div />
      </Styled(Row)>
      <Button
        action="Enviou form 'Criar a Senha'"
        disabled={true}
        id="cp_btn_submit"
        loading={false}
        onPress={[MockFunction]}
        type="button"
        value="Continuar"
      />
    </Col>
  </Styled(Row)>
</form>
`;

exports[`FirstAccess - Form Should render it when error is given 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Row
    center="xs"
  >
    <Styled(Col)>
      <strong>
        lembre-se que sua senha precisa ter ao menos:
      </strong>
    </Styled(Col)>
    <styled.ul>
      <RulePassword
        error="Inválida"
        id="min-caracter"
        rule={/\\.\\{8,15\\}/}
        text="Mínimo 8 e máximo 15 caracteres"
        value={undefined}
      />
      <RulePassword
        error="Inválida"
        id="um-maiscula"
        rule={/\\^\\(\\?=\\.\\*\\?\\[A-Z\\]\\)/}
        text="uma letra maiúscula"
        value={undefined}
      />
    </styled.ul>
    <styled.ul>
      <RulePassword
        error="Inválida"
        id="um-minuscula"
        rule={/\\(\\?=\\.\\*\\?\\[a-z\\]\\)/}
        text="Uma letra minúscula"
        value={undefined}
      />
      <RulePassword
        error="Inválida"
        id="carcter-especial"
        rule={/\\(\\?=\\.\\*\\?\\[#\\?!@\\$%\\^&\\*-\\]\\)/}
        text="Um caractere especial (Exemplo: @ # * & % $)"
        value={undefined}
      />
      <RulePassword
        error="Inválida"
        id="um-number"
        rule={/\\(\\?=\\.\\*\\?\\[0-9\\]\\)/}
        text="Um número"
        value={undefined}
      />
    </styled.ul>
  </Row>
  <styled.FormGroup
    error="Inválida"
    fieldId="cp_password"
    label="Criar uma Senha"
    showSuccess={false}
  >
    <TextInput
      eye={true}
      id="password"
      maskType={undefined}
      maxLength={15}
      name="password"
      onBlur={[MockFunction]}
      onChange={[Function]}
      passwordView={[Function]}
      placeholder="Insira aqui sua senha"
      value={undefined}
    />
  </styled.FormGroup>
  <styled.FormGroup
    error="Inválida"
    fieldId="cp_confirm_password"
    label="Confirmar senha"
    showSuccess={false}
  >
    <TextInput
      error="Inválida"
      eye={true}
      id="confirmPassword"
      maskType={undefined}
      maxLength={15}
      name="confirmPassword"
      onBlur={[MockFunction]}
      onChange={[Function]}
      passwordView={[Function]}
      placeholder="Repita a sua senha"
      value={undefined}
    />
  </styled.FormGroup>
  <Styled(Row)>
    <Col
      xs={12}
    >
      <ErrorText>
        Error
      </ErrorText>
    </Col>
  </Styled(Row)>
  <Styled(Row)
    center="md"
  >
    <Col
      md={12}
      xs={12}
    >
      <Styled(Row)>
        <UNDEFINED
          hl="pt-BR"
          onChange={undefined}
          onErrored={undefined}
          onExpired={undefined}
          sitekey=""
        />
        <div />
      </Styled(Row)>
      <Button
        action="Enviou form 'Criar a Senha'"
        disabled={true}
        id="cp_btn_submit"
        loading={false}
        onPress={[MockFunction]}
        type="button"
        value="Continuar"
      />
    </Col>
  </Styled(Row)>
</form>
`;
